In [1]:
#importing libraries
import pandas as pd
import numpy as np
import glob
import os


In [2]:
#set file path and glob for files
path = r"C:\Users\computer\Documents\GitHub\Data-Analysis-Project"

files = glob.glob(path + '/*.csv')
print(files)
#create empty list
lst = [] 

['C:\\Users\\computer\\Documents\\GitHub\\Data-Analysis-Project\\Abuja_Branch.csv', 'C:\\Users\\computer\\Documents\\GitHub\\Data-Analysis-Project\\Lagos_Branch.csv', 'C:\\Users\\computer\\Documents\\GitHub\\Data-Analysis-Project\\Port_Harcourt_Branch.csv']


In [3]:
#iterating through files
for file in files:
    os.path.join(path)
    temp_df = lst.append(file)

In [4]:
# creating the main data frame
df = pd.concat(map(pd.read_csv,files), ignore_index= True)
df.head()

Invoice ID Branch   City Customer type  Gender            Product line  \
0  692-92-5582      B  Abuja        Member  Female      Food and beverages   
1  351-62-0822      B  Abuja        Member  Female     Fashion accessories   
2  529-56-3974      B  Abuja        Member    Male  Electronic accessories   
3  299-46-1805      B  Abuja        Member  Female       Sports and travel   
4  319-50-3348      B  Abuja        Normal  Female      Home and lifestyle   

   Unit price  Quantity    Tax 5%      Total       Date   Time Payment  \
0     19742.4         3   2961.36   62188.56  2/20/2019  13:27    Card   
1      5212.8         4   1042.56   21893.76   2/6/2019  18:07    Epay   
2      9183.6         4   1836.72   38571.12   3/9/2019  17:03    Cash   
3     33739.2         6  10121.76  212556.96  1/15/2019  16:19    Cash   
4     14508.0         2   1450.80   30466.80  3/11/2019  15:30    Epay   

       cogs  gross margin percentage  gross income  Rating  
0   59227.2                 4.761905       2961.36     5.9  
1   20851.2                 4.761905       1042.56     4.5  
2   36734.4                 4.761905       1836.72     6.8  
3  202435.2                 4.761905      10121.76     4.5  
4   29016.0                 4.761905       1450.80     4.4

### 1. Data Exploration

In [5]:
#checking for the number of rows and columns
df.shape

(1000, 17)

In [6]:
#checking for null values
df.isnull().sum()

Invoice ID                 0
Branch                     0
City                       0
Customer type              0
Gender                     0
Product line               0
Unit price                 0
Quantity                   0
Tax 5%                     0
Total                      0
Date                       0
Time                       0
Payment                    0
cogs                       0
gross margin percentage    0
gross income               0
Rating                     0
dtype: int64

In [7]:
#checking for general data information
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 17 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Invoice ID               1000 non-null   object 
 1   Branch                   1000 non-null   object 
 2   City                     1000 non-null   object 
 3   Customer type            1000 non-null   object 
 4   Gender                   1000 non-null   object 
 5   Product line             1000 non-null   object 
 6   Unit price               1000 non-null   float64
 7   Quantity                 1000 non-null   int64  
 8   Tax 5%                   1000 non-null   float64
 9   Total                    1000 non-null   float64
 10  Date                     1000 non-null   object 
 11  Time                     1000 non-null   object 
 12  Payment                  1000 non-null   object 
 13  cogs                     1000 non-null   float64
 14  gross margin percentage  

##### From both the data info function and the isnull function, it can be seen that there are no missing values from the data gotten.

In [8]:
#see column names
df.columns

Index(['Invoice ID', 'Branch', 'City', 'Customer type', 'Gender',
       'Product line', 'Unit price', 'Quantity', 'Tax 5%', 'Total', 'Date',
       'Time', 'Payment', 'cogs', 'gross margin percentage', 'gross income',
       'Rating'],
      dtype='object')

In [9]:
df.describe()

Unit price     Quantity        Tax 5%          Total           cogs  \
count   1000.000000  1000.000000   1000.000000    1000.000000    1000.000000   
mean   20041.966800     5.510000   5536.572840  116268.029640  110731.456800   
std     9538.066205     2.923431   4215.177173   88518.720636   84303.543463   
min     3628.800000     1.000000    183.060000    3844.260000    3661.200000   
25%    11835.000000     3.000000   2132.955000   44792.055000   42659.100000   
50%    19882.800000     5.000000   4351.680000   91385.280000   87033.600000   
75%    28056.600000     8.000000   8080.290000  169686.090000  161605.800000   
max    35985.600000    10.000000  17874.000000  375354.000000  357480.000000   

       gross margin percentage  gross income      Rating  
count             1.000000e+03   1000.000000  1000.00000  
mean              4.761905e+00   5536.572840     6.97270  
std               6.131498e-14   4215.177173     1.71858  
min               4.761905e+00    183.060000     4.00000  
25%               4.761905e+00   2132.955000     5.50000  
50%               4.761905e+00   4351.680000     7.00000  
75%               4.761905e+00   8080.290000     8.50000  
max               4.761905e+00  17874.000000    10.00000

### Summary Statistics: 


1. The mean/average value for the cost of goods sold is greater than the median value; so it is positively skewed.
2. The maximum customer rating is 10 and the minimum customer rating is 1.
3. The maximum gross income is 17874.00 in naira, and the minimum gross income is 183.06 in naira.

### 2. Dealing with date time features.


In [10]:
df.head()

Invoice ID Branch   City Customer type  Gender            Product line  \
0  692-92-5582      B  Abuja        Member  Female      Food and beverages   
1  351-62-0822      B  Abuja        Member  Female     Fashion accessories   
2  529-56-3974      B  Abuja        Member    Male  Electronic accessories   
3  299-46-1805      B  Abuja        Member  Female       Sports and travel   
4  319-50-3348      B  Abuja        Normal  Female      Home and lifestyle   

   Unit price  Quantity    Tax 5%      Total       Date   Time Payment  \
0     19742.4         3   2961.36   62188.56  2/20/2019  13:27    Card   
1      5212.8         4   1042.56   21893.76   2/6/2019  18:07    Epay   
2      9183.6         4   1836.72   38571.12   3/9/2019  17:03    Cash   
3     33739.2         6  10121.76  212556.96  1/15/2019  16:19    Cash   
4     14508.0         2   1450.80   30466.80  3/11/2019  15:30    Epay   

       cogs  gross margin percentage  gross income  Rating  
0   59227.2                 4.761905       2961.36     5.9  
1   20851.2                 4.761905       1042.56     4.5  
2   36734.4                 4.761905       1836.72     6.8  
3  202435.2                 4.761905      10121.76     4.5  
4   29016.0                 4.761905       1450.80     4.4

In [11]:
#convert the Date column to datetime data type
df['Date'] = pd.to_datetime(df['Date'])

In [12]:
#confirm the data type for the date column
df.dtypes

Invoice ID                         object
Branch                             object
City                               object
Customer type                      object
Gender                             object
Product line                       object
Unit price                        float64
Quantity                            int64
Tax 5%                            float64
Total                             float64
Date                       datetime64[ns]
Time                               object
Payment                            object
cogs                              float64
gross margin percentage           float64
gross income                      float64
Rating                            float64
dtype: object

In [32]:
#retrieving the day,month,year and hour from the date and time columns
df['Day'] = df['Date'].dt.day
df['Month'] = df['Date'].dt.month
df['Year'] = df['Date'].dt.year
df['Time'] = pd.to_datetime(df['Time'])
df['Hour'] = df['Time'].dt.hour

In [33]:
df.head()

Invoice ID Branch   City Customer type  Gender            Product line  \
0  692-92-5582      B  Abuja        Member  Female      Food and beverages   
1  351-62-0822      B  Abuja        Member  Female     Fashion accessories   
2  529-56-3974      B  Abuja        Member    Male  Electronic accessories   
3  299-46-1805      B  Abuja        Member  Female       Sports and travel   
4  319-50-3348      B  Abuja        Normal  Female      Home and lifestyle   

   Unit price  Quantity    Tax 5%      Total  ...                Time Payment  \
0     19742.4         3   2961.36   62188.56  ... 2022-04-11 13:27:00    Card   
1      5212.8         4   1042.56   21893.76  ... 2022-04-11 18:07:00    Epay   
2      9183.6         4   1836.72   38571.12  ... 2022-04-11 17:03:00    Cash   
3     33739.2         6  10121.76  212556.96  ... 2022-04-11 16:19:00    Cash   
4     14508.0         2   1450.80   30466.80  ... 2022-04-11 15:30:00    Epay   

       cogs  gross margin percentage  gross income  Rating  Year  Day  Month  \
0   59227.2                 4.761905       2961.36     5.9  2019   20      2   
1   20851.2                 4.761905       1042.56     4.5  2019    6      2   
2   36734.4                 4.761905       1836.72     6.8  2019    9      3   
3  202435.2                 4.761905      10121.76     4.5  2019   15      1   
4   29016.0                 4.761905       1450.80     4.4  2019   11      3   

   Hour  
0    13  
1    18  
2    17  
3    16  
4    15  

[5 rows x 21 columns]

In [34]:
#check for unique hours
df['Hour'].unique()

array([13, 18, 17, 16, 15, 10, 12, 19, 14, 11, 20], dtype=int64)

### 3. Unique Values in Columns

In [36]:
df['Customer type'].unique()

array(['Member', 'Normal'], dtype=object)

In [37]:
df['Gender'].unique()

array(['Female', 'Male'], dtype=object)

In [38]:
df['Product line'].unique()

array(['Food and beverages', 'Fashion accessories',
       'Electronic accessories', 'Sports and travel',
       'Home and lifestyle', 'Health and beauty'], dtype=object)

In [39]:
df['Payment'].unique()

array(['Card', 'Epay', 'Cash'], dtype=object)